### PR TITLE
UN-325: Highlights (BE)

### PR DESCRIPTION
**Pull request by ababic**
Related ticket(s):
- https://national-archives.atlassian.net/browse/UN-325

## About these changes

Adds the model for `Highlight`, which includes a custom Django model field that functions like RecordChooserBlock.

As noted on the ticket, I've left out the field for linking to a "Closer Look / Record Revealed" page, as that page type doesn't exist yet.

## How to check these changes

Look for the Highlight item under the 'Snippets' menu option in Wagtail. You should be able to manage highlight instances successfully from there.

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly myself before handing over to reviewer.
- [x] Included the ticket number in the PR title to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [ ] Added/updated tests and documentation where relevant.

## For Reviewer

Once PR is merged, check and arrange to deploy on platform-sh.

### CODE REVIEW

**Review by jamesbiggs — COMMENTED** [review]
Hi Andy, this should bring my questions into the review :)

File: etna/highlights/models.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+
+    record = RecordChooserField(verbose_name=_("record"))
+
+    date_text = models.CharField(
+        verbose_name=_("date text"),
+        max_length=100,
+        help_text=_("Date(s) related to the record (max. character length: 100)"),
```
[the remark this answers is not in the document]
Comment (ababic) [reply etna/highlights/models.py:28]: This is a well known Django convention for marking hardcoded strings as 'translatable'. You might not need it now, but if you look at practically any third-party Django or Wagtail app, you'll see this used on most things user-facing. You can find out more on the topic here: https://docs.djangoproject.com/en/4.1/topics/i18n/translation/

File: etna/records/fields.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+from django.forms import ChoiceField
+
+from ..ciim.exceptions import KongBadRequestError
+from .models import Record
+from .widgets import RecordChooser
+
+
+class RecordChooserField(Field):
+    def __init__(self, *args, **kwargs):
+        kwargs["max_length"] = 50
```
[the remark this answers is not in the document]
Comment (ababic) [reply etna/records/fields.py]: The field stores the `iaid` of records in the database. The `max_length` is applied to the database column where that value is stored.

File: etna/highlights/models.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    )
+
+    description = RichTextField(verbose_name=_("description"))
```
[the remark this answers is not in the document]
Comment (ababic) [reply etna/highlights/models.py:31]: This is a well known Django convention for marking hardcoded strings as 'translatable'. You might not need it now, but if you look at practically any third-party Django or Wagtail app, you'll see this used on most things user-facing. You can find out more on the topic here: https://docs.djangoproject.com/en/4.1/topics/i18n/translation/

File: etna/highlights/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jamesbiggs) [reply etna/highlights/models.py:28]: Awesome! I didn't know there was something that could do this for an end-users language. I thought it was some sort of machine-level translation, for Django/Wagtail to understand. Thanks!

File: etna/records/fields.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jamesbiggs) [reply etna/records/fields.py]: Cool, thank you Andy 😃

**Review by jamesbiggs — APPROVED** [review]
Looks good, cheers Andy! 😃

**Pull request merged**